rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -313,6 +343,98 @@ fn bond_config_to_dbus(config: &BondConfig) -> HashMap<&str, zvariant::Value> {
     HashMap::from([("options", Value::new(options))])
 }
 
+fn bridge_config_to_dbus(bridge: &BridgeConfig) -> HashMap<&str, zvariant::Value> {
+    let mut hash = HashMap::new();
+
+    hash.insert("stp", bridge.stp.into());
+    if let Some(prio) = bridge.priority {
+        hash.insert("priority", prio.into());
+    }
+    if let Some(fwd_delay) = bridge.forward_delay {
+        hash.insert("forward-delay", fwd_delay.into());
+    }
+    if let Some(hello_time) = bridge.hello_time {
+        hash.insert("hello-time", hello_time.into());
+    }
+    if let Some(max_age) = bridge.max_age {
+        hash.insert("max-age", max_age.into());
+    }
+    if let Some(ageing_time) = bridge.ageing_time {
+        hash.insert("ageing-time", ageing_time.into());
+    }
+
+    hash
+}
+
+fn bridge_config_from_dbus(conn: &OwnedNestedHash) -> Option<BridgeConfig> {
+    let Some(bridge) = conn.get(BRIDGE_KEY) else {
+        return None;
+    };
+    let mut bc = BridgeConfig {
+        ..Default::default()
+    };
+
+    let Some(stp) = bridge.get("stp") else {
+        return None;
+    };
+    bc.stp = *stp.downcast_ref::<bool>()?;
+
+    if let Some(prio) = bridge.get("priority") {
+        bc.priority = Some(*prio.downcast_ref::<u32>()?);
+    }
+
+    if let Some(fwd_delay) = bridge.get("forward-delay") {
+        bc.forward_delay = Some(*fwd_delay.downcast_ref::<u32>()?);
```

Comment:
I know we have followed the same approach everywhere, but I wonder whether we want to return `None` if we fail to downcast one of those tiny fields. I have no opinion about that. But, sure, nothing to change in this PR.

Reply:
valid point, but we need to log the variable somehow, as it might be because of some API change?!
Another thing could be to use some macro in this regards. As it's also always the same but just exchanging the name in the hash and the name of the struct.

Reply:
Good idea, we could explore that path.